test_graph.py

```diff
@@ -76,10 +87,31 @@ def test_has_node(populate):
 
 
 def test_neighbors(populate):
-    """
-    Intermitent failures? Either the test is wrong or
-    my setup fixture isn't working as intended."""
     nodes = populate.nodes()
     node = nodes[0]
     edges = populate.neighbors(node)
     assert populate.g_dict[node] == edges
+
+
+def test_depth_trav(populate):
+    """This is weak. Having a hard time asserting on unknowns."""
```

Comment:
You are right, this test isn't the strongest. I would recommend hard coding in values, instead of using that awesome random generator you built. This way you know what is going in, and you can assert that is exactly what it should be.